File: src/backend/studio-reservation/src/main/java/org/taonaw/studio_reservation/usecase/command/reservation/reserveStudio/ReserveStudioCommand.java
package org.taonaw.studio_reservation.usecase.command.reservation.reserveStudio;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import org.taonaw.studio_reservation.domain.model.practiceTypeSetting.PracticeType;
import org.taonaw.studio_reservation.domain.model.reservation.UsageEquipments;
import org.taonaw.studio_reservation.domain.model.reservation.UsageTime;
import org.taonaw.studio_reservation.domain.model.reservation.UserCount;
import org.taonaw.studio_reservation.domain.model.reservation.UserInformation;
import org.taonaw.studio_reservation.domain.model.studio.StudioId;

@Getter
@Builder
@AllArgsConstructor
public class ReserveStudioCommand {
    private StudioId studioId;
    private UsageTime usageTime;
    private UserInformation userInformation;
    private UserCount userCount;
    private PracticeType practiceType;
    private UsageEquipments usageEquipments;
}

File: src/backend/reservation/src/main/kotlin/org/taonaw/reservation/usecase/reservation/reserveStudio/ReserveStudioCommand.kt
package org.taonaw.reservation.usecase.reservation.reserveStudio

import org.taonaw.reservation.domain.model.reservation.PracticeType
import org.taonaw.reservation.domain.model.reservation.UsageTime
import org.taonaw.reservation.domain.model.reservation.User
import org.taonaw.reservation.domain.model.reservation.UserCount
import org.taonaw.reservation.domain.model.reservation.rentalEquipment.RentalEquipments
import org.taonaw.reservation.domain.model.shared.StudioId

data class ReserveStudioCommand(
        val user: User,
        val studioId: StudioId,
        val usageTime: UsageTime,
        val userCount: UserCount,
        val practiceType: PracticeType,
        val rentalEquipments: RentalEquipments) {
}